code/drasil-code/Language/Drasil/Chunk/CodeDefinition.hs

```diff
@@ -15,53 +15,52 @@ import Control.Lens ((^.), makeLenses, view)
 -- | Finds the idea contained in the 'CodeChunk' used to make the 'CodeDefinition'.
-instance Idea         CodeDefinition where getA = getA . view cchunk
+instance Idea             CodeDefinition where getA = getA . view cchunk
 -- | Finds the 'Space' of the 'CodeChunk' used to make the 'CodeDefinition'.
-instance HasSpace     CodeDefinition where typ = cchunk . typ
+instance HasSpace         CodeDefinition where typ = cchunk . typ
 -- | Finds the 'Stage' dependent 'Symbol' of the 'CodeChunk' used to make the 'CodeDefinition'.
-instance HasSymbol    CodeDefinition where symbol c = symbol (c ^. cchunk)
+instance HasSymbol        CodeDefinition where symbol c = symbol (c ^. cchunk)
 -- | 'CodeDefinition's have a 'Quantity'.
-instance Quantity     CodeDefinition
+instance Quantity         CodeDefinition
 -- | Finds the code name of a 'CodeDefinition'.
 -- 'Function' 'CodeDefinition's are named with the function prefix to distinguish 
 -- them from the corresponding variable version.
-instance CodeIdea     CodeDefinition where 
+instance CodeIdea         CodeDefinition where 
   codeName (CD c@(CodeC _ Var) _ _ _) = codeName c
   codeName (CD c@(CodeC _ Func) _ _ _) = funcPrefix ++ codeName c
   codeChunk = view cchunk
 -- | Equal if 'UID's are equal.
-instance Eq           CodeDefinition where c1 == c2 = (c1 ^. uid) == (c2 ^. uid)
+instance Eq               CodeDefinition where c1 == c2 = (c1 ^. uid) == (c2 ^. uid)
 -- | Finds the units of the 'CodeChunk' used to make the 'CodeDefinition'.
-instance MayHaveUnit  CodeDefinition where getUnit = getUnit . view cchunk
+instance MayHaveUnit      CodeDefinition where getUnit = getUnit . view cchunk
 -- | Finds the defining expression of a CodeDefinition.
-instance DefiningExpr CodeDefinition where defnExpr = def
+instance DefiningCodeExpr CodeDefinition where codeExpr = def
+
+-- NOTE: We'll eventually want extra smart constructors that allow for custom
+--       CodeExprs inputs.
 
 -- | Constructs a 'CodeDefinition' where the underlying 'CodeChunk' is for a function.
 qtoc :: (Quantity q, DefiningExpr q, MayHaveUnit q) => q -> CodeDefinition
-qtoc q = CD (codeChunk $ quantfunc q) (q ^. defnExpr) [] Definition
+qtoc q = CD (codeChunk $ quantfunc q) (renderExpr $ q ^. defnExpr) [] Definition
 
 -- | Constructs a 'CodeDefinition' where the underlying 'CodeChunk' is for a variable.
 qtov :: QDefinition -> CodeDefinition
-qtov q = CD (codeChunk $ quantvar q) (q ^. defnExpr) [] Definition
+qtov q = CD (codeChunk $ quantvar q) (renderExpr $ q ^. defnExpr) [] Definition
 
 -- | Constructs a 'CodeDefinition' for an ODE.
 odeDef :: ODEInfo -> CodeDefinition
-odeDef info = CD (codeChunk $ quantfunc $ depVar info) (Matrix [odeSyst info]) 
-  (map ($ info) [tInit, tFinal, initVal, absTol . odeOpts, relTol . odeOpts, 
+odeDef info = CD (codeChunk $ quantfunc $ depVar info) (matrix [map renderExpr $ odeSyst info])
+  (map (renderExpr . ($ info)) [tInit, tFinal, initVal, absTol . odeOpts, relTol . odeOpts, 
```

Comment:
I'm not yet sure if we want to change `ODEInfo`s to use `CodeExpr`s instead of `Expr`s.
 https://github.com/JacquesCarette/Drasil/blob/5232d494995ca2d558dc60c2c81a8d4682a3c2e9/code/drasil-code/Language/Drasil/Data/ODEInfo.hs#L14-L23

Any thoughts on this?

Reply:
I would like to change it to use CodeExprs, but I'm not quite sure how the ODEs would use any objects as of right now.

Reply:
I think all of this can (and should) be changed to `CodeExpr`. If we need an ODE representation as an `Expr`, it will have to have its own definition there.

Reply:
Sounds good. I'll get this done and let you know when it's completed.

Reply:
Fixed -- I think it looks a lot better now. Thanks!